release-content/migration-guides/reflect_registration_changes.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Changes to type registration for reflection
+pull_requests: [15030, 20435, 20893]
+---
+
+Calling `.register_type` has long been a nuisance for Bevy users: both library authors and end users.
+This step was previously required in order to register reflected type information in the `TypeRegistry`.
+
+In Bevy 0.17 however, types which implement `Reflect` are now automatically registered, with the help of some compiler magic.
+This comes with a few caveats however:
+
+1. This functionality is gated by feature flags. Projects with default features off will not see reflected types unless one of these features is enabled.
```

Comment:
Do these features gate Reflection entirely? This sentence makes it sound like if you don't have auto-reflection on, you won't see any registered types in the registry, even if they were registered with `register_type`.

Reply:
They do not: only the autoregistration is gated.